src/api/utils/api_utils.py

```diff
@@ -368,4 +370,31 @@ def create_or_update_section_from_dict(section_dict, media=None):
 
     return section
 
+
+def create_unique_slug(title, model, field_name="slug", prefix=None):
+    """
+    Create a unique slug for a model instance based on the title.
+    """
+    if not title:
+        return None
+
+    slug = slugify(title)
+    if not model or not field_name:
+        return slug
+
+    if not model.objects.filter(**{field_name: slug}).exists():
+        return slug
+
+    if prefix:
+        prefixed_slug = f"{prefix}-{slug}"
+        if not model.objects.filter(**{field_name: prefixed_slug}).exists():
+            return prefixed_slug
+
+    number = 1
+    while True:
```

Comment:
I don't think I appreciate the way this is done. You can take advantage of the fact that timestamps are unique so you can use a datetimestamp instead of looping and trying to generate a random id each time. For example `20241009135980123` could suggest the slug name was created on the 9th of October 2024 at 13:59 PM. @abdullai-t